ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactTextInputManager.java

```diff
@@ -271,15 +271,17 @@ public void setAutoCapitalize(ReactEditText view, int autoCapitalize) {
 
   @ReactProp(name = "keyboardType")
   public void setKeyboardType(ReactEditText view, @Nullable String keyboardType) {
+    int flagsToUnset = 0;
     int flagsToSet = 0;
     if (KEYBOARD_TYPE_NUMERIC.equalsIgnoreCase(keyboardType)) {
-      flagsToSet = InputType.TYPE_CLASS_NUMBER;
+      flagsToUnset = InputType.TYPE_MASK_CLASS | InputType.TYPE_MASK_FLAGS;
```

Comment:
Maybe the InputType.TYPE_MASK_FLAGS doesn't need to be there, but I added it because I was using InputType.TYPE_NUMBER_FLAG_DECIMAL and didn't want something else that might been set to interfere with it, when they were ORed the other input type flags in `updateStagedInputTypeFlag`